File: mylibrary/src/main/java/com/diamondguide/library/ads/banner_ads/BannerBuilder.java
package com.diamondguide.library.ads.banner_ads;

import com.diamondguide.library.ads.admob.AdmobApi;
import com.diamondguide.library.ads.callback.BannerCallback;

import java.util.ArrayList;
import java.util.List;

public class BannerBuilder {
    private BannerCallback callBack = new BannerCallback();
    private final List<String> listId = new ArrayList<>();

    public BannerBuilder() {
    }

    public BannerBuilder setListId(List<String> listId) {
        this.listId.clear();
        this.listId.addAll(listId);
        return this;
    }

    public BannerBuilder setCallBack(BannerCallback callBack) {
        this.callBack = callBack;
        return this;
    }

    public BannerBuilder isIdApi() {
        this.listId.clear();
        this.listId.addAll(AdmobApi.getInstance().getListIDBannerAll());
        return this;
    }

    public void setListIdAd(String nameIdAd) {
        this.listId.clear();
        this.listId.addAll(AdmobApi.getInstance().getListIDByName(nameIdAd));
    }

    public BannerCallback getCallBack() {
        return callBack;
    }

    public List<String> getListId() {
        return listId;
    }
}

File: mylibrary/src/main/java/com/amazic/library/ads/banner_ads/BannerBuilder.kt
package com.amazic.library.ads.banner_ads

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout
import com.amazic.library.ads.callback.BannerCallback
import com.amazic.mylibrary.R
import com.google.android.libraries.ads.mobile.sdk.banner.AdView

class BannerBuilder {
    var callBack: BannerCallback = BannerCallback()
    var frContainer: FrameLayout? = null
    var useNewAdLoading: Boolean = false

    var bannerAdViewMain: AdView? = null
    var bannerAdViewSecondary: AdView? = null
    var bannerAdViewBackup: AdView? = null

    val listIdAdMain: MutableList<String> = mutableListOf()
    val listIdAdSecondary: MutableList<String> = mutableListOf()
    val listIdAdBackup: MutableList<String> = mutableListOf()

    var shimmerBanner: View? = null

    // Secondary constructor to match your first Java constructor
    constructor(frContainer: FrameLayout?) {
        this.frContainer = frContainer
    }

    // Secondary constructor to match your second Java constructor
    constructor(activity: Activity, frContainer: FrameLayout?, useNewAdLoading: Boolean) {
        this.useNewAdLoading = useNewAdLoading
        this.frContainer = frContainer

        // Show loading shimmer
        shimmerBanner = LayoutInflater.from(activity).inflate(R.layout.layout_shimmer_banner, null)
        frContainer?.addView(shimmerBanner)
    }

    // Builder-style pattern for Callback
    fun setCallBack(callBack: BannerCallback): BannerBuilder {
        this.callBack = callBack
        return this
    }

    fun setListIdAdMain(listIdAdMain: MutableList<String>): BannerBuilder {
        this.listIdAdMain.clear()
        this.listIdAdMain.addAll(listIdAdMain)
        return this
    }

    fun setListIdAdSecondary(listIdAdSecondary: MutableList<String>): BannerBuilder {
        this.listIdAdSecondary.clear()
        this.listIdAdSecondary.addAll(listIdAdSecondary)
        return this
    }

    fun setListIdAdBackup(listIdAdBackup: MutableList<String>): BannerBuilder {
        this.listIdAdBackup.clear()
        this.listIdAdBackup.addAll(listIdAdBackup)
        return this
    }
}